### PR TITLE
Process FOLLOWS_FROM spans in TraceView

### PR DESCRIPTION
**Pull request by rubenvp8510**
## Which problem is this PR solving?
- This fixes https://github.com/jaegertracing/jaeger-ui/issues/333

## Short description of the changes
- I noticed that there is FOLLOWS_FROM attributes on some spans that wasn't take into account when the indentation of the spans is computes on the TreeView.

Not sure if this is the proper way of handling this, I left two comments for two assumptions that I did. if some of the maintainers can review please. @tiffon @everett980 

Thanks!

### CODE REVIEW

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js [hunk 1]
```diff
@@ -52,14 +52,7 @@ export class UnconnectedSpanTreeOffset extends React.PureComponent<SpanTreeOffse
...
 
     this.ancestorIds = [];
-    let currentSpan: Span = props.span;
-    while (currentSpan) {
-      currentSpan = _get(_find(currentSpan.references, { refType: 'CHILD_OF' }), 'span');
-      if (currentSpan) {
-        this.ancestorIds.push(currentSpan.spanID);
-      }
-    }
-
```
Comment (tiffon) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js]: Looking at the code that builds the trace tree, which is then turned into the array of spans with parents preceding children (and children are sorted by startTime):

https://github.com/jaegertracing/jaeger-ui/blob/38efa49e96d34cb666fa232da14346ceccd896d4/packages/jaeger-ui/src/selectors/trace.js#L72-L75

Looks like the algorithm (which could probably be better) is to use the span ID of the first reference, as long as that reference is FOLLOWS_FROM or CHILD_OF.

In accumulating the ancestor IDs, we should aim to be consistent with that process so it accurately reflects the order the list.

Therefore, maybe the following would work?

```jsx
while (currentSpan && Array.isArray(currentSpan.references)) {
  const { (refType, span } = currentSpan.references[0] || {};
  if (span && (refType === 'FOLLOWS_FROM' || refType === 'CHILD_OF')) {
    this.ancestorIds.push(span.spanID);
    currentSpan = span;
  }
}
```

What do you think? 

@everett980 Can you weigh in?

Comment (everett980) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js]: I think the precedent for handling reference is not particularly strong. AFAIK there is nothing preventing a span from being `FOLLOWS_FROM` multiple other spans. With regards to indentation it should only indent once (based on depth) but I do not know how the UI handles this case in other ways. Does the span with two `FOLLOWS_FROM` show up twice in the list, once per reference?

Purely from a UI layout perspective I like the implementation in your comment. However, I am uncertain if we should have better support for multiple references.

Comment (yurishkuro) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js]: Multiple references are rarely used within the same trace (i.e. to represent a JOIN), the more realistic use case is something like multiple traces coalescing into a single batch write operation, which causally depends on all of the previous traces. However, the _first_ reference in the list is always used to pick the trace ID for the new span (actually, different Jaeger clients might be different in that regard, e.g.. they may give priority to the _first child-of reference_).

**Comment by rubenvp8510** [comment]
I like the algorithm you put in the comments, simpler that mine, at the beginning I thought we will give an special treatment to the FOLLOWS_FROM, that is why I split it in two different things, but after an analysis of my implementation and  your code it seems like from the UI perspective (or at least for this view) we won't do a distinction. Correct me if this is wrong.

I'll simplify my code.

**Comment by rubenvp8510** [comment]
I also think that we need to put some tests for this, to avoid regressions.

**Comment by rubenvp8510** [comment]
@tiffon @everett980 I added one test to take FOLLOWS_FROM into account. I think this is ready for another review, if there is no more comments and you decide it's ok, then it is ready for merge.

**Pull request merged**